### PR TITLE
Don't fail trying to read boot image wihtout enable

### PR DESCRIPTION
**Pull request by Qalthos**
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

cliconf/eos

### CODE REVIEW

File: lib/ansible/plugins/cliconf/eos.py [hunk 1]
```diff
@@ -247,10 +247,15 @@ def get_device_info(self):
 
         device_info['network_os_hostname'] = data['hostname']
 
-        reply = self.get('bash timeout 5 cat /mnt/flash/boot-config')
-        match = re.search(r'SWI=(.+)$', reply, re.M)
-        if match:
-            device_info['network_os_image'] = match.group(1)
+        try:
+            reply = self.get('bash timeout 5 cat /mnt/flash/boot-config')
```
Comment (Qalthos) [inline lib/ansible/plugins/cliconf/eos.py:251]: Is there a specific reason we do this instead of `show boot-config | json`?

**Comment by pabelanger** [comment]
This was per-merged tested on zuul.ansible.com and appears to fix the issue:

  https://logs.zuul.ansible.com/48/48/75114ee6342481028f1bda3ab449e339dba00a42/check/ansible-test-network-integration-eos-devel/ace067e/job-output.txt

File: lib/ansible/plugins/cliconf/eos.py [hunk 2]
```diff
@@ -247,10 +247,15 @@ def get_device_info(self):
...
+
+            match = re.search(r'SWI=(.+)$', reply, re.M)
+            if match:
+                device_info['network_os_image'] = match.group(1)
+        except AnsibleConnectionFailure:
+            # This requires enable mode to run
```
Comment (ganeshrn) [inline lib/ansible/plugins/cliconf/eos.py:257]: Can we emit a warning/info instead of silently failing

**Review by ganeshrn — APPROVED** [review]

File: lib/ansible/plugins/cliconf/eos.py (hunk 2 above)
Comment (Qalthos) [reply lib/ansible/plugins/cliconf/eos.py:257]: I think a warning might be a bit strong for this, but I don't know what the implications of this data not being populated are.

**Pull request merged**